src/main/java/com/alvarium/annotators/vulnerability/OsvApiHandler.java

```diff
@@ -32,15 +32,15 @@ public class OsvApiHandler implements VulnerabilityApiHandler {
     final private Gson gson = new Gson();
     final private String endpoint;
 
-    public OsvApiHandler() {
-        this.endpoint = "https://api.osv.dev/v1/querybatch";
+    public OsvApiHandler(VulnerabilityAnnotatorConfig cfg) {
+        this.endpoint = cfg.getProvider().uri();
     }
 
     private OsvBatchResponse makeApiCall(Map<String,String> packages) throws VulnerabilityException {
         try {
             // Create the HTTP client and POST request
             CloseableHttpClient httpClient = HttpClients.createDefault();
-            HttpPost httpPost = new HttpPost(this.endpoint);
+            HttpPost httpPost = new HttpPost(this.endpoint + "/v1/querybatch"); 

```

Comment:
Shouldn't `/v1/querybatch` be coming from the config? If that route changes or is upgraded it will necessitate a rewrite, recompile, redeploy of the app instead of just a config change.

Reply:
This one was a bit tricky for me because on one hand  the `ServiceProvider` implementation adds the port at the end of the URL so the endpoint path would have been malformed (i.e., `https://example.com/v1/querybatch:443` instead of `https://example.com:443/v1/querybatch`). On the other hand, I didn't want to overcomplicate the service provider implementation just for this one annotator.

The approach I used eventually in fed5b82 is adding new entries for these paths and constructing the complete path from the annotator config class
```json
    {
      "kind": "vulnerability",
      "type": "osv",
      "paths": {
        "queryBatch": "/v1/querybatch",
        "query": "/v1/query"
      },
      "provider": {
        "host": "example.com",
        "port": 80,
        "protocol": "https"
      }
    }
```

```java
public class VulnerabilityAnnotatorConfig extends AnnotatorConfig {
    
    ...

    public String getQueryBatchPath() {
        return this.provider.uri() + this.paths.getBatchQuery();
    }

    public String getQueryPath() {
        return this.provider.uri() + this.paths.getQuery();
    }
}
```